plugins/foreman_discovery/4.0/index.md

```diff
@@ -0,0 +1,739 @@
+---
+layout: plugin
+title: Foreman Discovery 3.0 Manual
+version: 3.0
+---
+
+# 1. {{ page.title }}
+
+This plugin enables Foreman to do automatic bare-metal discovery of unknown
+nodes on the provisioning network. New nodes self-register into Foreman and
+upload facts collected by Facter (serial id, network interfaces, memory,
+disks). The registered nodes show up on Discovered Hosts page and provisioning
+can be initiated either manually (via UI/CLI or API) or automatically via
+predefined Discovery Rules.
+
+All communication is through Foreman Proxy which is running a plugin and has
+direct access both to the provisioning network and Foreman instance:
+
+    --
+    Discovered Host > Foreman Proxy > Foreman
+    --
+    Foreman > Foreman Proxy > Discovered Host
+    --
+
+## 1.1 Release Notes
+
+For each Foreman version there is particular version of Foreman Discovery
+plugin:
+
+<table class="table table-bordered table-condensed">
+  <tr>
+    <th>Foreman version</th>
+    <th>Plugin version</th>
+    <th>Proxy version</th>
+    <th>Image version</th>
+    <th>CLI version</th>
+  </tr>
+  <tr>
+    <td>&lt;= 1.2</td>
+    <td>1.0.2</td>
+    <td>N/A</td>
+    <td>N/A</td>
+    <td>N/A</td>
+  </tr>
+  <tr>
+    <td>= 1.3</td>
+    <td>1.1.0</td>
+    <td>N/A</td>
+    <td>0.1.1</td>
+    <td>N/A</td>
+  </tr>
+  <tr>
+    <td>= 1.4</td>
+    <td>1.2.0</td>
+    <td>N/A</td>
+    <td>0.3.0</td>
+    <td>N/A</td>
+  </tr>
+  <tr>
+    <td>= 1.5</td>
+    <td>1.3.0</td>
+    <td>N/A</td>
+    <td>0.5.0</td>
+    <td>N/A</td>
+  </tr>
+  <tr>
+    <td>= 1.6</td>
+    <td>1.4.0</td>
+    <td>N/A</td>
+    <td>0.6 or 2.0</td>
+    <td>N/A</td>
+  </tr>
+  <tr>
+    <td>= 1.7</td>
+    <td>2.0.0</td>
+    <td>0.1</td>
+    <td>2.0</td>
+    <td>0.0.1</td>
+  </tr>
+  <tr>
+    <td>= 1.8</td>
+    <td>3.0.0</td>
+    <td>1.0.2</td>
+    <td>2.1</td>
+    <td>0.0.1</td>
+  </tr>
+  <tr>
+    <td>= 1.9</td>
+    <td>4.0.0</td>
+    <td>1.0.2</td>
+    <td>2.1.1</td>
+    <td>0.0.2</td>
+  </tr>
+</table>
+
+### 1.1.1 Foreman Discovery plugin
+
+**4.0**: Release notes
+
+A bugfixing release, we had issues with facter compatibility, rebooting when
```

Comment:
New fields, status indicators, reboot all hosts, improved API etc?  Up to you though.